sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -62,12 +85,20 @@ export const imdsMsi: MSI = {
     clientId?: string,
     getTokenOptions?: GetTokenOptions
   ): Promise<boolean> {
+    // if the PodIdenityEndpoint environment variable was set no need to probe the endpoint, it can be assumed to exist
+    if (process.env.AZURE_POD_IDENTITY_AUTHORITY_HOST) {
+      return true;
+    }
+
     const { span, updatedOptions: options } = createSpan(
       "ManagedIdentityCredential-pingImdsEndpoint",
       getTokenOptions
     );
 
-    const requestOptions = prepareRequestOptions(resource, clientId);
+    const requestOptions = prepareRequestOptions(resource, clientId, {
+      skipMetadataHeader: true,
```

Comment:
Looks like the header is redundantly deleted below (line 107).

Reply:
oh thanks! Copy-paste issue from https://github.com/Azure/azure-sdk-for-js/pull/16875